src/maker.rs

```diff
@@ -78,38 +78,100 @@ impl Maker {
         self.mid_market_rate = None;
     }
 
-    pub fn update_bitcoin_balance(&mut self, balance: bitcoin::Amount) {
-        self.btc_balance = balance;
+    pub fn update_bitcoin_balance(
+        &mut self,
+        balance: bitcoin::Amount,
+    ) -> anyhow::Result<Option<BtcDaiOrder>> {
+        // if we had a balance and the balance did not change => no new orders
+        if let Some(previous_balance) = self.btc_balance {
+            if previous_balance == balance {
+                return Ok(None);
+            }
+        }
+
+        // TODO: Evaluate if we even care... we could also just "spam" with creating new orders and simplify this logic (add tests once we decide)
+
+        // we know that the balance changed and have to take in account the max_sell_amount
+        match (self.btc_max_sell_amount, self.btc_balance) {
+            (Some(max_sell_amount), Some(previous_balance))
+                if balance >= max_sell_amount && !(previous_balance < max_sell_amount) =>
```

Comment:
We are duplicating the logic of taking in account the `max_sell_amount` (the logic is also present when creating a new order).

What if instead, we just create a new order and compare it to the last order created? If it's the same amount (because the max_sell_amount was capping the order anyway and the rate did not change) then we returned `None`, other returned the order created.

Reply:
Reduced the logic to always publishing an order if the balance' value changed. 
Recorded possible follow-up item for this.